internal/common/slogger/logrus.go

```diff
@@ -0,0 +1,52 @@
+package slogger
+
+import (
+	"strings"
+
+	"github.com/osbuild/osbuild-composer/pkg/jobqueue"
+	"github.com/sirupsen/logrus"
+)
+
+type simpleLogrus struct {
+	logger *logrus.Logger
+}
+
+func NewLogrusLogger(logger *logrus.Logger) jobqueue.SimpleLogger {
+	return &simpleLogrus{logger: logger}
+}
+
+func (s *simpleLogrus) log(level logrus.Level, err error, msg string, args ...string) {
+	if len(args)%2 != 0 {
+		panic("log arguments must be even (key value pairs)")
+	}
+	var fields = make(logrus.Fields, len(args)/2+1)
+	var sb strings.Builder
+	sb.WriteString(msg)
+	if len(args) > 0 || err != nil {
+		sb.WriteString(":")
+	}
+	for i := 0; i < len(args); i += 2 {
+		k := args[i]
+		v := args[i+1]
+		fields[k] = v
```

Comment:
I want to be extra careful not to drop any bit of information in case you run some configurations without structured logging (e.g. text-based). Your surprise reveals, however, that this is unlikely the case, removing :-)